lib/msal-browser/src/interaction_handler/PopupHandler.ts

```diff
@@ -65,145 +63,35 @@ export class PopupHandler extends InteractionHandler {
             const maxTicks = timeout / BrowserConstants.POLL_INTERVAL_MS;
             let ticks = 0;
 
-            const intervalId = setInterval(() => {
-                if (popupWindow.closed) {
-                    // Window is closed
-                    this.cleanPopup();
-                    clearInterval(intervalId);
-                    reject(BrowserAuthError.createUserCancelledError());
-                    return;
-                }
-
-                let href: string = Constants.EMPTY_STRING;
-                try {
-                    /*
-                     * Will throw if cross origin,
-                     * which should be caught and ignored
-                     * since we need the interval to keep running while on STS UI.
-                     */
-                    href = popupWindow.location.href;
-                } catch (e) {}
-
-                // Don't process blank pages or cross domain
-                if (StringUtils.isEmpty(href) || href === "about:blank") {
-                    return;
-                }
-
-                // Only run clock when we are on same domain
-                ticks++;
-                const contentHash = popupWindow.location.hash;
-                if (UrlString.hashContainsKnownProperties(contentHash)) {
-                    // Success case
-                    this.cleanPopup(popupWindow);
-                    clearInterval(intervalId);
-                    resolve(contentHash);
-                    return;
-                } else if (ticks > maxTicks) {
-                    // Timeout error
-                    this.cleanPopup(popupWindow);
-                    clearInterval(intervalId);
-                    reject(BrowserAuthError.createMonitorPopupTimeoutError());
-                    return;
-                }
-            }, BrowserConstants.POLL_INTERVAL_MS);
+            this.popupUtils.monitorPopupForSameOrigin(popupWindow).then(() => {
```

Comment:
Once we are on the same origin, can we just return the hash/url, instead of polling again?

Reply:
Wasn't sure why the hash extraction was inside the polling previously so didn't want to change it. But I've pulled it out now.